lib/useCache.ts

```diff
@@ -75,7 +76,12 @@ class PersistantCache implements Cache {
 
   *write(key: string, data: unknown) {
     const location = new URL(`./${key}.jsonl`, this.location);
-    yield* call(() => ensureFile(location));
+    if (!existsSync(dirname(location.pathname))) {
+      mkdirSync(dirname(location.pathname), { recursive: true });
+    }
+    if (!existsSync(location)) {
+      writeFileSync(location.pathname, "");
+    }
```

Comment:
The sync operations are slow. Have you tried running it after this? I'm worried that it might be too slow. Have you looked at the stat and not using sync operations?